mne/gui/_file_traits.py

```diff
@@ -285,14 +287,40 @@ def _get_n_omitted(self):
     @cached_property
     def _get_inst(self):
         if self.file:
-            info = read_info(self.file, verbose=False)
-            if info['dig'] is None:
+            try:
+                info = read_info(self.file, verbose=False)
+            except ValueError:  # try reading dig_montage
```

Comment:
That will clutter the log with an error message in the other case where there is no error to be concerned about. I think returning an info dict with the montage would be a logical extension of `read_info` (only change a case that was not functional previously), but if you don't want to change `read_info`, can you read the header of the file to determine what kind of file it is and treat it accordingly?